toonz/sources/toonzqt/schematicviewer.cpp

```diff
@@ -432,6 +544,198 @@ void SchematicSceneViewer::showEvent(QShowEvent *se) {
   }
 }
 
+//------------------------------------------------------------------
+
+void SchematicSceneViewer::enterEvent(QEvent *e) {
+  switch (m_cursorMode) {
+  case CursorMode::Hand:
+    setToolCursor(this, ToolCursor::PanCursor);
+    break;
+  case CursorMode::Zoom:
+    setToolCursor(this, ToolCursor::ZoomCursor);
+    break;
+  default:
+    setToolCursor(this, ToolCursor::StrokeSelectCursor);
+    break;
+  }
+}
+
+//------------------------------------------------------------------
+
+void SchematicSceneViewer::leaveEvent(QEvent *e) { setCursor(Qt::ArrowCursor); }
+
+//------------------------------------------------------------------
+
+void SchematicSceneViewer::tabletEvent(QTabletEvent *e) {
+  // qDebug() << "[tabletEvent]";
+  if (e->type() == QTabletEvent::TabletPress) {
+    m_stylusUsed = e->pointerType() ? true : false;
+  } else if (e->type() == QTabletEvent::TabletRelease) {
+    m_stylusUsed = false;
+  }
+
+  e->accept();
+}
+
+//------------------------------------------------------------------
+
+void SchematicSceneViewer::gestureEvent(QGestureEvent *e) {
+  // qDebug() << "[gestureEvent]";
+  m_gestureActive = false;
+  if (QGesture *swipe = e->gesture(Qt::SwipeGesture)) {
+    m_gestureActive = true;
+  } else if (QGesture *pan = e->gesture(Qt::PanGesture)) {
+    m_gestureActive = true;
+  }
+  if (QGesture *pinch = e->gesture(Qt::PinchGesture)) {
+    QPinchGesture *gesture = static_cast<QPinchGesture *>(pinch);
+    QPinchGesture::ChangeFlags changeFlags = gesture->changeFlags();
+    QPoint firstCenter = mapFromGlobal(gesture->centerPoint().toPoint());
+
+    if (gesture->state() == Qt::GestureStarted) {
+      m_gestureActive = true;
+    } else if (gesture->state() == Qt::GestureFinished) {
+      m_gestureActive = false;
+      m_zooming       = false;
+      m_scaleFactor   = 0.0;
+    } else {
+      if (changeFlags & QPinchGesture::ScaleFactorChanged) {
+        double scaleFactor = gesture->scaleFactor();
+        // the scale factor makes for too sensitive scaling
+        // divide the change in half
+        if (scaleFactor > 1) {
+          double decimalValue = scaleFactor - 1;
+          decimalValue /= 1.5;
+          scaleFactor = 1 + decimalValue;
+        } else if (scaleFactor < 1) {
+          double decimalValue = 1 - scaleFactor;
+          decimalValue /= 1.5;
+          scaleFactor = 1 - decimalValue;
+        }
+        if (!m_zooming) {
+          double delta = scaleFactor - 1;
+          m_scaleFactor += delta;
+          if (m_scaleFactor > .2 || m_scaleFactor < -.2) {
+            m_zooming = true;
+          }
+        }
+        if (m_zooming) {
+          changeScale(firstCenter * getDevPixRatio(), scaleFactor);
```

Comment:
Well, calling `setMouseTracking(true)` in ctor and modifying this line to
`changeScale(m_oldWinPos, scaleFactor);` 
seems to resolve the offset. 

I think I need to investigate further, anyway.